### PR TITLE
Fix FTP unable to retrieve server IP address in tethering mode

### LINKED ISSUE #1664: Crash on starting FTP server on Android 9.0 when device is running as hotspot
Symptom found on LG Nexus 5x running AOSPExtended (9.0).

1. Switch off wifi
2. Switch on hotspot
3. Start Amaze
4. Open drawer
5. Tap FTP server
6. A toast saying "Error getting local inet address" popped up
7. Start FTP server
8. Amaze and FTP server crashed

Possibly reason is the behaviour change of `WifiManager` in Android Q causing problem getting hotspot's IP address (i.e. 192.168.43.1). More details to follow after getting the device back.

However, judging the scale of the changes required, I think the fix will happen past 3.3.3 release.

### PR DESCRIPTION
**Pull request by bowiechen**
Fixes #1664.

In Android 9+, Google introduced code to randomize host's IP address in wifi tethering mode. The wifi tethering is still in the same subnet, however. See:
https://android.googlesource.com/platform/frameworks/base/+/0f27eed8f666e5ee38cea16b7feb2fd8e43c9b9e

### CODE REVIEW

**Comment by EmmanuelMess** [comment]
Was a bit ill, checking this PRs now.

**Comment by EmmanuelMess** [comment]
Ok, can't test, as I can't create a hotspot in AVD with API 28 or 29.

**Comment by VishalNehra** [comment]
Just got android 10 update for my Oneplus 6t. Will test this.

**Comment by TranceLove** [comment]
Wow, never thought it could be this easy to fix - was thinking of changing method calls to the official way that Android introduced to request hotspot using [android.net.wifi.WifiManager.startLocalOnlyHotspot()](https://developer.android.com/reference/android/net/wifi/WifiManager#startLocalOnlyHotspot(android.net.wifi.WifiManager.LocalOnlyHotspotCallback,%20android.os.Handler))... but as long as it worked, it is good.

Tested on LG Nexus 5x running AOSPExtended (9.0), FTP service can be started in hotspot mode without crashing Amaze. I'll vote +1.

**Comment by bowiechen** [comment]
@TranceLove Yep, I was also surprised to realize this issue was this easy to fix. I was going down the documentation rabbit hole until I came across [a Reddit post](https://old.reddit.com/r/GooglePixel/comments/8vypfb).

File: app/src/main/java/com/amaze/filemanager/asynchronous/services/ftp/FtpService.java [hunk 1]
```diff
@@ -324,7 +321,7 @@ public static InetAddress getLocalInetAddress(Context context) {
                 while (addresses.hasMoreElements()) {
                     InetAddress address = addresses.nextElement();
 
-                    if(WIFI_AP_ADDRESS.equals(address.getHostAddress()) && isEnabledWifiHotspot(context))
+                    if(address.getHostAddress().startsWith(WIFI_AP_ADDRESS_PREFIX) && isEnabledWifiHotspot(context))
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/asynchronous/services/ftp/FtpService.java]: Please add a null check for address

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/asynchronous/services/ftp/FtpService.java]: Please also add null check on FtpNotification#92 where the original crash happens.

**Comment by VishalNehra** [comment]
Tested and working.

**Comment by TranceLove** [comment]
I wonder you don't need to merge current `master` into your PR but a rebase should already do?

**Comment by bowiechen** [comment]
@TranceLove The merge commit was generated by GitHub to resolve a conflict on two of the `strings.xml`. I wonder if I can rebase locally to eliminate the need for a merge commit.

**Pull request merged**